### DISCUSSION TITLE
Standard PHP Extensions?

### DISCUSSION THREAD

**Discussion by lonnietc** [roadrunner-server/roadrunner#1573] [Q&A] [answered]
Hello,

After a lot of research on PHP servers for a project that I am starting on, I came across RoadRunner which seems to have potential to meet my simple needs (hopefully).

The basic problem that I am initially testing out is the ability to take an existing PHP web application and have RoadRunner be able to serve it properly.

The question that I have is regarding PHP extensions.  From what I understand, and have read, RoadRunner does not need them (?) but my web application requires:

PHP 7.x
MySQL 5.6+ or MariaDB 10.0+

Intl PHP Extension
OpenSSL PHP Extension
PDO PHP Extension
Mbstring PHP Extension
XML PHP Extension
Ctype PHP Extension
JSON PHP Extension
PHP GD Extension

I think that the first 2 (above) are not a problem, but I am not sure how to deal with the others, if possible at all.

If I can get these figured out then my second effort will be to embed the PHP web application in an embedded RoadRunner application, but that will be later.

Any help or guidance would be greatly appreciated.
Thanks and have a great day.

**Comment by rustatian** [comment, marked as answer]
Hey @lonnietc and nice to meet you 👋🏻
RR is a standalone application (like PHP-FPM) and doesn't require any of the PHP extensions. Moreover, RR doesn't limit you in writing your code, you can use any existing extension you want. The only 1 requirement is that you'll need a worker (just PHP code), you can find an example here: https://roadrunner.dev/docs/php-worker/2023.x/en
